modules/config.xqm

```diff
@@ -12,6 +12,10 @@ declare namespace system="http://exist-db.org/xquery/system";
 declare namespace expath="http://expath.org/ns/pkg";
 declare namespace repo="http://exist-db.org/xquery/repo";
 
+
+declare variable $config:login-domain := "org.exist.public-repo.login";
+declare variable $config:base-url := "/exist/apps/public-repo/";
```

Comment:
This assumes a lot about the server and app, and should be composed of other constants.

Reply:
Well, yes. I thought it is at least better to have it here among the configurations. But in the end this is likely a function that checks for is-proxied and the context as well.

Reply:
The reason I went for a simple string is that the repo will in almost all cases have exactly one URL it will be requested from.

Reply:
If you have a suggestion which variables can be used to construct the base-url more reliably I am all ears!

Reply:
- `/exist` is available via `request:get-context-path()`. 
- `/apps` is available via `$exist:prefix`
- `public-repo` is available via `$exist:controller`

In other words, the components hardcoded here in `$config:base-url` are actually dependent 3 system variables: [the Jetty webapp context](https://github.com/eXist-db/exist/blob/develop/exist-jetty-config/src/main/resources/org/exist/jetty/etc/webapps/exist-webapp-context.xml#L5-L6), the [root controller](https://github.com/eXist-db/exist/blob/develop/exist-jetty-config/src/main/resources/webapp/WEB-INF/controller-config.xml#L45), and [the app's target collection](https://github.com/eXist-db/public-repo/blob/master/repo.xml#L10).

At present, the safest way to gather this data is from within the `controller.xql`, which uniquely has access to the `$exist:*` variables, but from there you can simply provide the needed value as a parameter to the views that need to construct a base-url.

Reply:
What do you think about the current solution:
Use the base url constructed in the controller by adding as a parameter to all pages.
That way any further logic, like `is-proxied` can be implemented in a central location.

Reply:
Yes, I think the current solution is good. The parameter is added to all pages as needed.